### PR TITLE
Fix typo in `_ManifestEvalVisitor.visit_equal`

### PR DESCRIPTION
**Pull request by ddrinka**
@Fokko here's another runtime bug I'm seeing while playing with this new Manifest evaluator (#5845)

### CODE REVIEW

File: python/pyiceberg/expressions/visitors.py [hunk 1]
```diff
@@ -526,7 +526,7 @@ def visit_equal(self, term: BoundTerm, literal: Literal[Any]) -> bool:
         if lower > literal.value:
             return ROWS_CANNOT_MATCH
 
-        upper = _from_byte_buffer(term.ref().field.field_type, field.lower_bound)
+        upper = _from_byte_buffer(term.ref().field.field_type, field.upper_bound)
```
Comment (ajantha-bhat) [inline python/pyiceberg/expressions/visitors.py:529]: good finding 👍🏻. 

nit: maybe good to have unit tests around these.

**Comment by Fokko** [comment]
Thanks for spotting this one @ddrinka It looks like we also need a not-`None` check. I also noticed that this is being fixed in https://github.com/apache/iceberg/pull/6069. So I'll close this one for now. If you need this right away, feel free to open up a new PR (with the check included).

**Comment by rdblue** [comment]
I reopened this because I think it's a good idea to get it in independently. Thanks for finding this, @ddrinka!

**Comment by Fokko** [comment]
Fixed the linting issue, thanks @ddrinka!

**Pull request merged**